sphinx_search/_static/js/rtd_sphinx_search.js

```diff
@@ -71,97 +75,333 @@ const createDomNode = (nodeName, attributes) => {
+        sectionData.highlight["sections.title"] !== undefined &&
+        sectionData.highlight["sections.title"].length >= 1
+    ) {
+        section_subheading = sectionData.highlight["sections.title"][0];
+    }
+
+    let section_content =
+        sectionData._source.content.substring(0, 100) + " ...";
+
+    if (
+        sectionData.highlight["sections.content"] !== undefined &&
+        sectionData.highlight["sections.content"].length >= 1
+    ) {
+        section_content =
+            "... " + sectionData.highlight["sections.content"][0] + " ...";
+    }
+
+    let section_link = `${page_link}#${sectionData._source.id}`;
+
+    let section_id = "hit__" + COUNT;
+
+    let section_html = $u.template(section_template, {
+        section_link: section_link,
+        section_id: section_id,
+        section_subheading: section_subheading,
+        section_content: section_content
+    });
+
+    return section_html;
+};
+
+/**
+ * Generate and return html structure
+ * for a sphinx domain result.
+ *
+ * @param {Object} domainData object containing the result data
+ * @param {String} page_link link of the main page. It is used to construct the section link
+ */
+const get_domain_html = (domainData, page_link) => {
+    let domain_template =
+        '<a href="<%= domain_link %>"> \
+            <div class="outer_div_page_results" id="<%= domain_id %>"> \
+                <span class="search__result__subheading"> \
+                    <%= domain_subheading %> \
+                </span> \
+                <p class="search__result__content"><%= domain_content %></p> \
+            </div> \
+        </a> \
+        <br class="br-for-hits">';
+
+    let domain_link = `${page_link}#${domainData._source.anchor}`;
+    let domain_role_name = domainData._source.role_name;
+    let domain_type_display = domainData._source.type_display;
+    let domain_doc_display = domainData._source.doc_display;
+    let domain_display_name = domainData._source.display_name;
+    let domain_name = domainData._source.name;
+
+    // take values from highlighted fields (if present)
+    if (domainData.highlight !== undefined && domainData.highlight !== null) {
+        if (
+            domainData.highlight["domains.name"] !== undefined &&
+            domainData.highlight["domains.name"].length >= 1
+        ) {
+            domain_name = domainData.highlight["domains.name"][0];
+        }
+
+        if (
+            domainData.highlight["domains.display_name"] !== undefined &&
+            domainData.highlight["domains.display_name"].length >= 1
+        ) {
+            domain_display_name =
+                domainData.highlight["domains.display_name"][0];
+        }
+
+        if (
+            domainData.highlight["domains.type_display"] !== undefined &&
+            domainData.highlight["domains.type_display"].length >= 1
+        ) {
+            domain_type_display =
+                domainData.highlight["domains.type_display"][0];
+        }
+    }
```

Comment:
Can we write a function to abstract these 3 functions that do the same thing? `get_highlight_list_data('domains.display_name`)` or similar?